discovery-core/src/main/java/io/micronaut/discovery/cloud/digitalocean/DigitalOceanMetadataResolver.java

```diff
@@ -138,11 +138,11 @@ public Optional<ComputeInstanceMetadata> resolve(Environment environment) {
             }
         } catch (MalformedURLException mue) {
             if (LOG.isErrorEnabled()) {
-                LOG.error("Digital Ocean metadataUrl value is invalid!: " + configuration.getUrl(), mue);
+                LOG.error("Digital Ocean metadataUrl value is invalid!: {}", configuration.getUrl(), mue);
             }
         } catch (IOException ioe) {
             if (LOG.isErrorEnabled()) {
-                LOG.error("Error connecting to" + configuration.getUrl() + "reading instance metadata", ioe);
+                LOG.error("Error connecting to{}reading instance metadata", configuration.getUrl(), ioe);
```

Comment:
In this case the recipe run brought to light an issue with how the message is composed.
```suggestion
                LOG.error("Error connecting to {} reading instance metadata", configuration.getUrl(), ioe);
```